### ISSUE TITLE
<Menu.Item /> throws "Unable to find node on an unmounted component"

### ISSUE THREAD

**Issue by Robertmw** [ant-design/ant-design#17016]
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://vprdr.codesandbox.io)

### Steps to reproduce
- Open console
- Navigate thru pages

### What is expected?
No errors should be thrown while navigating

### What is actually happening?
React throws "Uncaught Invariant Violation: Unable to find node on an unmounted component. "

| Environment | Info |
|---|---|
| antd | 3.19.2 |
| React | 16.8.6 |
| System | macOS |
| Browser | Chrome 74.0.3729.169 |

---
Stack trace: https://imgur.com/bOtZ1JQ

**Comment by zombieJ** [comment]
Not get the error. Could you help to give more detail about the reproduce step or give a gif?

**Comment by yoonwaiyan** [comment]
I can reproduce the error in CodeSandbox as provided by OP. I have the same issue in my antd version 2.x .


![Jun-10-2019 17-28-21](https://user-images.githubusercontent.com/3081772/59186334-2baaf680-8ba5-11e9-81f7-cb428c8968a2.gif)

**Comment by zombieJ** [comment]
ref: https://github.com/facebook/react/issues/14188

**Comment by zombieJ** [comment]
Create related issue with React: https://github.com/facebook/react/issues/15857

**Comment by GreenHandLittleWhite** [comment]
@zombieJ
这个[例子](https://github.com/facebook/react/issues/14188#issuecomment-500478400)上再套一层可以用，虽然目的好像不一样了
```
<React.Suspense fallback={<p>Loading</p>}>
    {routes.map((el, index) => (
        <Route
            key={index}
            path={el.path}
            exact={el.exact}
            component={el.component}
        />
    ))}
</React.Suspense>
```

**Comment by Robertmw** [comment]
This should be fixed in [react v16.9.0](https://github.com/facebook/react/releases/tag/v16.9.0) - https://github.com/facebook/react/pull/15312

If you guys can update the dependencies before next release it will be much appreciated  :D

**Comment by afc163** [comment]
Fixed in react@16.9.
